index.js

```diff
@@ -0,0 +1,95 @@
+'use strict';
+
+const Joi = require('joi');
+const emailer = require('./email');
+
+// Joi Schema Validation
+const SCHEMA_ADDRESS = Joi.string().email();
+const SCHEMA_ADDRESSES = Joi.array()
+    .items(SCHEMA_ADDRESS)
+    .min(0);
+const SCHEMA_STATUS = Joi.string();
+const SCHEMA_STATUSES = Joi.array()
+    .items(SCHEMA_STATUS)
+    .min(0);
+const SCHEMA_BUILD_SETTINGS = Joi.alternatives().try(
+    Joi.object().keys({ addresses: SCHEMA_ADDRESSES, statuses: SCHEMA_STATUSES }),
+    Joi.string().email()
+    );
+const SCHEMA_BUILD_DATA = Joi.object()
+    .keys({
+        settings: SCHEMA_BUILD_SETTINGS,
+        status: SCHEMA_STATUS
+    });
+const SCHEMA_SMTP_CONFIG = Joi.object()
+    .keys({
+        host: Joi.string(),
+        port: Joi.number().integer(),
+        from: Joi.string().email()
+    });
+
+const DESCRIPTION_MAP = {
+    SUCCESS: 'Everything looks good!',
+    FAILURE: 'Did not work as expected.',
+    ABORTED: 'Aborted mid-flight',
+    RUNNING: 'Testing your code...',
+    QUEUED: 'Looking for a place to park...'
+};
+const DEFAULT_STATUSES = ['FAILURE'];
+
+let config;
+let server;
+let eventName;
+
+class EmailNotifier {
+    constructor(inConfig, inServer, inEventName) {
+        config = Joi.attempt(inConfig, SCHEMA_SMTP_CONFIG.unknown(true),
+            'Invalid config for email notifications');
+        server = inServer;
+        eventName = inEventName; // The event that notify() will trigger a listener on
+    }
+
+    /**
+    * Sets listener on server event of name 'eventName'
+    * Currently, event is triggered with a build status is updated
+    * @method notify
+    * @return {Promise} resolves to false if status is not in notification statuses
+    *                   resolves to emailer if status is in notification statuses
+    */
+    notify() {
+        return new Promise((resolve) => {
+            server.on(eventName, (buildData) => {
```

Comment:
non-blocking: it might be helpful to describe what the `buildData` is for this event. I only suggest this because if the module changes in the future and `buildData` is different, it would help the future bug-fixers understand what needs to be addressed